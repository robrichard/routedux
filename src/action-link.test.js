import ActionLink from './action-link';
import {mount} from 'enzyme';
import ezJson from 'enzyme-to-json';


it("dispatches an action on click", () => {

  // given
  const store = {
    pathForAction: jest.fn(() => '/my/path'),
    dispatch: jest.fn()
  };
  const props = {
    action: {type: 'ACTION', id: '123'},
    children: "Hello World!"
  };
  const context = {store};

  const wrapper = mount(ActionLink(props, context));
  // when
  wrapper.simulate('click');

  //then
  expect(store.pathForAction.mock.calls).toEqual([[ {type:'ACTION', id: '123'} ]]);
  expect(store.dispatch.mock.calls).toEqual([[ {type: 'ACTION', id: '123'} ]]);

});

it("renders the url calculated by our internal function", () => {
  // given
  const store = {
    pathForAction: jest.fn(() => '/my/path'),
    dispatch: jest.fn()
  };
  const props = {
    action: {},
    children: "Hello World!"
  };
  const context = {store};

  const wrapper = mount(ActionLink(props, context));

  expect(ezJson(wrapper)).toMatchSnapshot();

});

it("additional props are passed through", () => {
  // given
  const store = {
    pathForAction: jest.fn(() => '/my/path'),
    dispatch: jest.fn()
  };
  const props = {
    action: {},
    children: "Hello World!",
    className: "foo"
  };
  const context = {store};

  const wrapper = mount(ActionLink(props, context));

  expect(ezJson(wrapper)).toMatchSnapshot();

});
