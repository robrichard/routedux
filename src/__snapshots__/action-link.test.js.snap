// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`additional props are passed through 1`] = `
<a
  className="foo"
  href="/my/path"
  onClick={[Function]}
>
  Hello World!
</a>
`;

exports[`renders the url calculated by our internal function 1`] = `
<a
  href="/my/path"
  onClick={[Function]}
>
  Hello World!
</a>
`;
